tests/ui/unused_unit.rs

```diff
@@ -31,6 +29,20 @@ impl Into<()> for Unitter {
     }
 }
 
+trait Trait {
+    fn redundant<F: FnOnce() -> (), G, H>(&self, _f: F, _g: G, _h: H)
+    where
+        G: FnMut() -> (),
+        H: Fn() -> ();
+}
+
+impl Trait for Unitter {
+    fn redundant<F: FnOnce() -> (), G, H>(&self, _f: F, _g: G, _h: H)
+    where
+        G: FnMut() -> (),
+        H: Fn() -> () {}
```

Comment:
This should not trigger in trait implementations, since the implementer may not have influence on the trait definition.

Reply:
Yep, I thought about this too, but in this case I think it's OK because the definition does not change, it's just expressed differently (both cases return nothing)

Reply:
Oh yeah, you can leave out the unit return in the implementation, even if it is in the declaration. :+1: